test/cluster/basic_test.go

```diff
@@ -101,3 +103,424 @@ func TestBasicCluster(t *testing.T) {
+			require.NoError(t, c.ShutdownServer(c.Servers[currentLeader]))
+		}
+		if oldLeader != -1 {
+			require.NoError(t, c.StartServer(c.Servers[oldLeader]))
+		}
+		oldLeader = currentLeader
+	}
+
+	//make sure all 3 nodes are active
+	require.Eventually(t, func() bool {
+		clusterStatusResEnv, err = c.Servers[currentLeader].QueryClusterStatus(t)
+		return clusterStatusResEnv != nil && len(clusterStatusResEnv.GetResponse().GetActive()) == 3
+	}, 30*time.Second, 100*time.Millisecond)
+
+	var dataEnv *types.GetDataResponseEnvelope
+	for i := 0; i < 3; i++ {
+		require.Eventually(t, func() bool {
+			dataEnv, err = c.Servers[currentLeader].QueryData(t, worldstate.DefaultDBName, strconv.Itoa(i))
+			return dataEnv != nil && dataEnv.GetResponse().GetValue() != nil && err == nil
+		}, 30*time.Second, 100*time.Millisecond)
+
+		dataVal := dataEnv.GetResponse().GetValue()
+		t.Logf("data: %+v", dataVal)
+
+		require.Equal(t, []byte{uint8(i)}, dataVal)
+	}
+}
+
+//Scenario:
+// - start 3 servers in a cluster.
+// - wait for one to be the leader.
+// - stop the leader and one follower.
+// - submit a tx to the remaining server.
+// - there is no majority to peak leader => tx fails.
+// - restart one server so now there will be a majority to peak a leader.
+// - wait for one to be the leader.
+// - submit a tx => tx accepts.
+func TestNoMajorityToChooseLeader(t *testing.T) {
+	dir, err := ioutil.TempDir("", "int-test")
+	require.NoError(t, err)
+
+	nPort, pPort := getPorts(3)
+	setupConfig := &setup.Config{
+		NumberOfServers:     3,
+		TestDirAbsolutePath: dir,
+		BDBBinaryPath:       "../../bin/bdb",
+		CmdTimeout:          10 * time.Second,
+		BaseNodePort:        nPort,
+		BasePeerPort:        pPort,
+		CheckRedirectFunc: func(req *http.Request, via []*http.Request) error {
+			return errors.Errorf("Redirect blocked in test client: url: '%s', referrer: '%s', #via: %d", req.URL, req.Referer(), len(via))
+		},
+	}
+	c, err := setup.NewCluster(setupConfig)
+	require.NoError(t, err)
+	defer c.ShutdownAndCleanup()
+
+	require.NoError(t, c.Start())
+
+	var clusterStatusResEnv *types.GetClusterStatusResponseEnvelope
+	leaders := [3]string{"", "", ""}
+	for i, srv := range c.Servers {
+		require.Eventually(t, func() bool {
+			clusterStatusResEnv, err = srv.QueryClusterStatus(t)
+			return clusterStatusResEnv != nil && clusterStatusResEnv.GetResponse().GetLeader() != "" && err == nil
+		}, 30*time.Second, 100*time.Millisecond)
+		leaders[i] = clusterStatusResEnv.GetResponse().GetLeader()
+	}
+
+	require.Equal(t, leaders[0], leaders[1])
+	require.Equal(t, leaders[0], leaders[2])
+
+	leaderServer, leaderVal := c.GetServerByID(leaders[0])
+	require.NotEqual(t, leaderVal, -1)
+	require.NotNil(t, leaderServer)
+
+	follower1 := (leaderVal + 1) % 3
+	follower2 := (leaderVal + 2) % 3
+
+	require.NoError(t, c.ShutdownServer(c.Servers[leaderVal]))
+	require.NoError(t, c.ShutdownServer(c.Servers[follower1]))
+
+	time.Sleep(10 * time.Second)
```

Comment:
get status and wait for eventually - only one node is active, also check leader is ""

Reply:
Done